app/src/main/java/com/cp/tasky/auth/login/presentation/LoginScreen.kt

```diff
@@ -109,7 +103,7 @@ private fun LoginScreen(
                     }
                 }
 
-                if (networkState is Result.Success) {
+                if (viewState.isLoggedIn) {
                     onLoginSuccess()
                 }
```

Comment:
You're having a side-effect here (executing a non-Compose function inside a Composable). To avoid calling this more often than intended, wrap this into LaunchedEffect:
```
LaunchedEffect(viewState.isLoggedIn) {
    if(viewState.isLoggedIn) {
        onLoginSuccess()
    }
}
```
Also something to keep in mind is that if you work with such states for navigation, you might need to reset these after handling them. In this case not an issue, since you'll clear the login screen after logging in, but if it could stay on the back stack and wouldn't reset it, you'd be stuck in an infinite navigate loop when trying to navigate back to this screen (since the isLoggedIn boolean will remain true and therefore instantly perform the navigation).

I personally rather work with Channels for such one-time events. Working with states is also fine, but needs the above considerations.